### PR TITLE
Remove unreachable error branch in memset calls (and in repeat)

### PR DESCRIPTION
**Pull request by jakobnissen**
Some places use the pattern memset(A, v, length(A)), which requires a conversion UInt(length(A)). This is technically fallible, but can't actually fail when A is a Memory or Array.
Remove the dead error branch by casting to UInt instead.

Similarly, in repeat(x, r), r is first checked to be nonnegative, then converted to UInt, then used in multiple calls where it is converted to UInt each time. Here, only do it once.

### CODE REVIEW

File: base/strings/string.jl [hunk 1]
```diff
@@ -570,9 +570,10 @@ julia> repeat('A', 3)
 ```
 """
 function repeat(c::AbstractChar, r::Integer)
+    r < 0 && throw(ArgumentError("can't repeat a character $r times"))
```
Comment (Seelengrab) [inline base/strings/string.jl:573]: Unrelated change?

Comment (Seelengrab) [reply base/strings/string.jl:573]: Ah, this is mentioned - can this be added to the PR title?

**Review by Seelengrab — APPROVED** [review]

**Comment by jakobnissen** [comment]
Test failures are unrelated

**Review by IanButterworth — APPROVED** [review]

**Comment by oscardssmith** [comment]
It does seem like it would be good if we had a better way of expressing this info to the compiler...
